packages/@aws-cdk/tmp-toolkit-helpers/src/api/refactoring/index.ts

```diff
@@ -0,0 +1,415 @@
+import * as crypto from 'crypto';
+import { createHash } from 'node:crypto';
+import type { TypedMapping } from '@aws-cdk/cloudformation-diff';
+import {
+  formatTypedMappings as fmtTypedMappings,
+  formatAmbiguousMappings as fmtAmbiguousMappings,
+} from '@aws-cdk/cloudformation-diff';
+import type * as cxapi from '@aws-cdk/cx-api';
+import type { StackSummary } from '@aws-sdk/client-cloudformation';
+import { deserializeStructure } from '../../util';
+import type { SdkProvider } from '../aws-auth';
+import { Mode } from '../plugin';
+import { StringWriteStream } from '../streams';
+
+interface CloudFormationTemplate {
```

Comment:
Surely we have this definition somewhere else already?

Reply:
Seriously, I wonder if we shouldn't invest in a proper CloudFormation Template parser at some point... it kinda bothers me that we do ad-hoc JSON processing all over the place :/ but at every point it's the path of least resistance, and never worth doing anything more... resulting in loads of duplication